src/Compilers/CSharp/Test/Emit/Emit/EntryPointTests.cs

```diff
@@ -107,6 +107,32 @@ public class D
                 Diagnostic(ErrorCode.ERR_MultipleEntryPoints, "Main"));
         }
 
+        [Fact]
+        public void ERR_MultipleEntryPoints_SyncAndAsync()
+        {
+            string source = @"
+using System.Threading.Tasks;
+
+public class C
+{
+  public static async Task Main() { await Task.Delay(1); }
+  public static void Main(string[] a) { System.Console.WriteLine(2); }
+}
+
+public class D
+{
+  public static string Main() { System.Console.WriteLine(3); return null; }
+}
+";
+            var compilation = CompileConsoleApp(source);
+
+            compilation.VerifyDiagnostics(
+                // (12,24): warning CS0028: 'D.Main()' has the wrong signature to be an entry point
+                Diagnostic(ErrorCode.WRN_InvalidMainSig, "Main").WithArguments("D.Main()"),
+                // (4,28): error CS0017: Program has more than one entry point defined. Compile with /main to specify the type that contains the entry point.
+                Diagnostic(ErrorCode.ERR_MultipleEntryPoints, "Main"));
```

Comment:
@RikkiGibson @333fred 
Isn't the error message for CS0017 a bit misleading?
The error can occur when multiple entry points are defined in the **same** type. How come that "specifying the type that contains the entry point through -main option" help?

Reply:
For this particular case, yeah that's a bit misleading. Seems like the cleanest thing would be to introduce a new error message for this particular situation. To do so:
1. Add the message to CSharpResources.resx. Follow the pattern of the other errors in that file, it should be something like `ERR_ShortishName`.
2. Add an error with exact same name to ErrorCodes.cs.

Reply:
@333fred Thanks. I'll create a new error for that case after fixing the current test failure.
Do you know under what lang version the test run by default, and how to change it?

Reply:
@Youssef1313 tests run under latest by default. To change it, pass one of the `TestOptions.RegularX` values to `CreateCompilation`.